### DISCUSSION TITLE
Please consider preparing ubuntu/debian packages

### DISCUSSION THREAD

**Discussion by Pheromon** [mhx/dwarfs#103] [Ideas]
it would be so much more easier to use this tool!

**Comment by adminx01** [comment]
You can use MPR
git clone https://mpr.makedeb.org/dwarfs-bin.git && cd dwarfs-bin && makedeb -si

Debian has outdated FUSE anyway, you would better switch to arch based.

**Reply by Pheromon** [reply]
debian has fuse 2.9.9 which is the same version of arch.
Or is it another package you are talking about?

**Reply by daci12345** [reply]
fuse3 is outdated on debian not fuse2, but the mpr package should work with any version of debian 11 and above, and ubuntu 20 and above

**Reply by Pheromon** [reply]
Ok I see, but we are talking about release 3.10.5 against 3.11, the difference is minimal: the only important change is "Add support for flag FOPEN_NOFLUSH for avoiding flush on close" that should completely be useless for this project.

So, I don't see how this can be a blocking factor

**Comment by mhx** [comment]
I only have marginal experience with debian/ubuntu (although admittedly I'm using ubuntu vms to build the static release binaries).

I'd definitely appreciate if someone would look into maintaining packages for these distros, but I'm unlikely going to be that person. :)

I'm moving this to discussions for now.

**Reply by hexahigh** [reply]
I'm a little late to this discussion, but I'm currently [hosting an apt repo](https://apt.080609.xyz) which has dwarfs in it.

**Reply by alexmyczko** [reply]
@hexahigh don’t you want to maintain things for official debian? i would be glad to review/sponsor. all you need is one of: a) mentors.debian.net account/dput stuff here
b) put your stuff on any webserver (what appears to be the case now)
c) create salsa.debian.org account

**Reply by hexahigh** [reply]
I recently switched away from Debian and my experience with Debian's build system has been pretty bad, so I'm definitely not the person who should be maintaining it.

**Comment by silentnoodlemaster** [comment]
[coming to experimental](https://ftp-master.debian.org/new/dwarfs_0.10.0-1.html)

**Reply by alexmyczko** [reply]
i hope it will be accepted soon, and makes it into debian trixie! as soon as it is accepted to experimental, update and upload to sid is needed…

**Reply by mhx** [reply]
Just curious, has there been an update in the meantime?

**Reply by silentnoodlemaster** [reply]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1068852

A year ago it got the pending tag and nothing has happened since

**Reply by alexmyczko** [reply]
it was rejected and requires another upload...

**Reply by alexmyczko** [reply]
help is needed:

dwarfs_0.13.0+ds-1_amd6-
s REJECTED
'debian/copyright' lists 'Copyright: Unknown' for
'folly/folly/external/fast-crc32/*'.
The files themselves just say '/* Generated by https://github.com/corsix/fast-crc32/*/'
no l
• * MIT licensed*, but don't name copyright holder. Since the Expat license requires reproducing "the above copyright notice",
, we need to know who that is.
Please look up the actual copyright holder from the upstream
https://github.com/corsix/fast-crc32/ project and update 'debian/copyright'
Separately, a few files under 'fbthrift/thrift/lib/java/
'are copyrighted by
VMware rather than Meta - specifically
'ForkJoinPoolScheduler.java'
'ForkJoinPoolSchedulerTest.java', and
'AbstractSchedulerTest.java'
, all in
'fbthrift/thrift/lib/java/runtime/./resources/'. They say
'Copyright (c) 2017-2021 VMware Inc. or its affiliates' and are Apache-2.0
licensed. The 'fbthrift/*' stanza in 'debian/ copyright' currently only lists
Meta Platforms as copyright holder, so VMware needs to be added there too.

to whoever reads this and wants to help, reach out to tarzeau on irc or #debian-mentors happy to share deb src package and review/sponsor as many uploads as needed.

**Reply by mhx** [reply]
@alexmyczko, I'm in the process of (mostly) de-meta-ing DwarFS. I've forked the (very minimal) bits I need from fbthrift (since I was planning on extending them and upstream hasn't been responding to my PRs for more than half a year) and am actively removing any dependency on folly. This should hopefully help with issues like this (and save my time keeping folly/fbthrift up-to-date when I only need less than 5% of it).

**Reply by mhx** [reply]
Another thing that should hopefully help with copyright/licensing: https://github.com/mhx/dwarfs/blob/mhx/thrift-lite/REUSE.toml

**Reply by mhx** [reply]
v0.15.0 has all the changes mentioned above.
